src/Microsoft.AspNet.Mvc.Core/Resources.resx

```diff
@@ -460,4 +457,7 @@
   <data name="TempData_CannotSerializeDictionary" xml:space="preserve">
     <value>The '{0}' cannot serialize a dictionary with a key of type '{1}' to session state.</value>
   </data>
+  <data name="ValueInterfaceAbstractOrOpenGenericTypesCannotBeActivated" xml:space="preserve">
+    <value>Value, Interface, Abstract or Open Generic types cannot be activated by '{0}'.</value>
```

Comment:
:+1: was going to suggest something similar. We could even simplify it further and say that 'non-instantiable types and value types' aren't supported.